src/main/java/uk/gov/hmcts/reform/pip/subscription/management/repository/SubscriptionRepository.java

```diff
@@ -26,10 +26,24 @@ public interface SubscriptionRepository extends JpaRepository<Subscription, Long
 
     void deleteById(UUID id);
 
-    @Query(value = "SELECT * FROM Subscription WHERE search_type = :search_type AND search_value = :search_value",
+    @Query(value = "SELECT * FROM Subscription "
+        + "WHERE search_type = :search_type "
+        + "AND search_value = :search_value "
+        + "AND :search_type <> 'LOCATION_ID'",
         nativeQuery = true)
     List<Subscription> findSubscriptionsBySearchValue(@Param("search_type") String searchType,
                                                       @Param("search_value") String searchValue);
 
+
+    @Query(value = "SELECT * FROM Subscription "
+        + "WHERE search_type = :search_type "
+        + "AND search_value = :search_value "
+        + "AND :search_type = 'LOCATION_ID' "
+        + "AND (ARRAY_LENGTH(list_type, 1) IS NULL OR (list_type && string_to_array(:list_type, ',')))",
```

Comment:
Just to check would that also return if the array is empty as well as null? Or does it need to be a null array